Support for OSX

davidbrochart opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

@conda-forge-admin, please rerender
<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

davidbrochart commented:
@conda-forge-admin, please rerender

davidbrochart commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to re-render for you, but it looks like there was nothing to do.

chrisburr commented on recipe/meta.yaml:
```diff
@@ -1,34 +1,38 @@
+    - swig 4.0.1
   host:
-    - python
+    - python 3
```
Pinning the Python version like this will make bad things happen. My other comment for how to do it correctly.
```suggestion
    - python
```

chrisburr replied:
To add, you should ask the bot to rerender after fixing this.

chrisburr commented on recipe/meta.yaml:
```diff
@@ -1,34 +1,38 @@
 build:
   number: {{ build }}
-  skip: true  # [not linux]
+  skip: true  # [win]
```
```suggestion
  skip: true  # [win or py2k]
```

chrisburr commented on recipe/meta.yaml:
```diff
@@ -1,34 +1,38 @@
+    - openmp
   run:
-    - python
+    - python 3
```
```suggestion
    - python
```

davidbrochart commented:
Thanks @chrisburr, I made the changes.

chrisburr commented:
@conda-forge-admin, please rerender

martinRenou commented:
I suppose you did try this, but just in case I am adding this comment concerning osx versions: https://stackoverflow.com/a/14285922/9734668

davidbrochart commented:
@Benatti1991 did we change something in the installation recently? I cannot import pychrono even on Linux now: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=118813&view=logs&jobId=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=6d4b912b-175d-51da-0fd9-4d30fe1eb4e7

Benatti1991 commented:
If you are using the master branch no, nothing changed.

davidbrochart commented:
@chrisburr when I had `python` (with no version) and `py2k`, I couldn't even `import pychrono` (`not found`). In the last commit I added `python 3` back and removed `py2k` and everything works. Do you know what could be the issue?

chrisburr commented:
The issue is that the package is being installed in the wrong place `pythonX.Ym/site-packages` instead of `pythonX.Y/site-packages`. It passes with `python 3` as it builds with Python 3.8 and the pymalloc build now uses the same ABI and doesn't have the `m` suffix, see [here](https://docs.python.org/3/whatsnew/3.8.html#build-and-c-api-changes).

The issue with the current recipe is that it says to conda "the same binaries work with all Python versions" which definitely won't be true for any compiled package. The only reason there are two builds is because you haven't rerendered.

Tangentially related, it seems the Python 3.8 migration PR was closed (#5) so you should add [this file](https://github.com/regro-cf-autotick-bot/pychrono-feedstock/blob/48c08838ef39149bbd7212089b75b073d3cb79c3/.ci_support/migrations/python38.yaml) and rerender to enable Python 3.8 and well as 3.6 and 3.7.

davidbrochart commented:
I don't understand, I do handle the `m` suffix according to the Python version, do you mean that it should always install in `pythonX.Y/site-packages` regardless of the Python version?

davidbrochart commented:
@conda-forge-admin, please rerender

isuruf commented on recipe/meta.yaml:
```diff
@@ -1,34 +1,38 @@
   host:
     - python
+    - eigen
+    - openmp
```
See https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp

davidbrochart replied:
Thanks @isuruf, I will add that in the next commit.

davidbrochart commented:
@Benatti1991 we're almost there, the last error is for python 3.6 on OSX, with a `double B0;` and `#define B0 0`. Maybe you can help: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=119056&view=logs&jobId=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=113b264d-9551-5065-da10-7eb3bdeaf1bc

davidbrochart commented:
I created a PR to fix the compilation error we have on OSX for Python 3.6: https://github.com/projectchrono/chrono/pull/222
